lib/utils/addon-utils.ts

```diff
@@ -47,6 +47,39 @@ export function dependable(...addOns: string[]) {
       return result;
     };
 
+    return descriptor;
+  };
+}
+
+/**
+ * Decorator function that accepts a list of AddOns and
+ * throws error if those addons are scheduled to be added as well
+ * As they should not be deployed with
+ * @param addOns 
+ * @returns 
+ */
+export function conflictsWith(...addOns: string[]) {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  return function (target: Object, key: string | symbol, descriptor: PropertyDescriptor) {
+    const originalMethod = descriptor.value;
+
+    descriptor.value = function( ...args: any[]) {
+      // const dependencies: (Promise<Construct> | undefined)[] = [];
+      const clusterInfo: ClusterInfo = args[0];
+      const stack = clusterInfo.cluster.stack.stackName;
+
+      addOns.forEach( (addOn) => {
+        const dep = clusterInfo.getScheduledAddOn(addOn);
+        if (dep){
+          throw new Error(`Deploying ${stack} failed due to conflicting add-on: ${addOn}.`)
+        }
+      });
+
+      const result: Promise<Construct> = originalMethod.apply(this, args);
+
+      return result;
```

Comment:
minor: why do we need this result? why not `return originalMethod.apply ...`

Reply:
Fixed it to return without assigning.